app/src/main/java/com/android/voyageur/ui/trip/AddActivity.kt

```diff
@@ -84,12 +79,27 @@ fun AddActivityScreen(tripsViewModel: TripsViewModel, navigationActions: Navigat
 
     val today = normalizeToMidnight(Date())
     val dateNormalized = activityDate?.let { normalizeToMidnight(Date(it)) } ?: Date(0)
+    val selectedTrip = tripsViewModel.selectedTrip.value!!
+
+    if (activityDate != null &&
+        (dateNormalized.after(normalizeToMidnight(selectedTrip.endDate.toDate())) ||
+            dateNormalized.before(normalizeToMidnight(selectedTrip.startDate.toDate())))) {
```

Comment:
Consider normalizing selectedTrip.startDate and selectedTrip.endDate before the conditional check. This will enhance code readability by reducing redundant calls to normalizeToMidnight and make the date comparison logic more straightforward.

Reply:
Also, following Andreea's and Ioana's comments, you could add a null check for selectedTrip before accessing its properties. This will prevent potential NullPointerException errors and ensure the code handles cases where selectedTrip may be null gracefully. I'll also leave the final decision up to you.